<launch>
    <arg name="world" default="maze"/>
    <arg name="robot_count" default="2"/>
    <arg name="run" default="0"/>
    <arg name="method" default="gvg"/>
    <arg name="bs_pose" default="0,0"/>
    <arg name="environment"  default=''/>

    <param name="use_sim_time" value="true" />
    <node name="Stage" pkg="stage_ros" type="stageros" args="$(find gvgexploration)/world/$(arg world).world">
        <param name="base_watchdog_timeout" value="0" />
    </node>
    <group ns="robot_0">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="1"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="0" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="6" />
            <arg name="relay_robots" value="1,2,3,4,5" />
        </include>

    </group>

    <group ns="robot_1">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="2"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="1" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="6" />
            <arg name="relay_robots" value="0,2,3,7,5" />
        </include>
    </group>

    <group ns="robot_2">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="3"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="2" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="6" />
            <arg name="relay_robots" value="0,1,3,4,5" />
        </include>
    </group>

    <group ns="robot_3">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="4"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="3" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="6" />
            <arg name="relay_robots" value="0,1,2,4,5" />
        </include>
    </group>

    <group ns="robot_4">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="5"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="4" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="6" />
            <arg name="relay_robots" value="0,1,2,3,5" />
        </include>
    </group>

    <group ns="robot_5">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="6"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="5" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="6" />
            <arg name="relay_robots" value="0,1,2,3,4" />
        </include>
    </group>

    <group ns="robot_6">
        <node pkg="nav2d_navigator" type="explore_client" name="explore_client"/>
        <include file="$(find gvgexploration)/launch/nodes.launch">
            <arg name="robot_name" value="7"/>
        </include>
        <include file="$(find recurrent_connectivity)/launch/robot.launch">
            <arg name="robot_name" value="6" />
            <arg name="robot_type" value="1" />
            <arg name="relay_robots" value="0,1,2,3,4,5" />
        </include>
    </group>

    <node name="roscbt" pkg="recurrent_connectivity" type="roscbt.py">
        <rosparam file="$(find recurrent_connectivity)/param/roscbt_config$(arg robot_count).yaml"/>
    </node>

    <node name="localizer" pkg="gvgexploration" type="$(arg world)_localize.sh" />
    <node name="map_analyzer" pkg="gvgexploration" type="map_analyzer.py"/>

    <node name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find recurrent_connectivity)/param/recurrent_connectivity.rviz" output="screen" />

</launch>

